tests/model_fields_/test_embedded_model.py

```diff
@@ -144,23 +146,360 @@ def setUpTestData(cls):
             for x in range(6)
         ]
 
-    def test_exact(self):
-        self.assertCountEqual(Holder.objects.filter(data__integer=3), [self.objs[3]])
+    def test_exact_expr(self):
+        with self.assertNumQueries(1) as ctx:
+            self.assertCountEqual(Holder.objects.filter(data__integer=Value(4) - 1), [self.objs[3]])
+        query = ctx.captured_queries[0]["sql"]
+        self.assertAggregateQuery(
+            query,
+            "model_fields__holder",
+            [
+                {
+                    "$match": {
+                        "$expr": {
+                            "$eq": [
+                                {"$getField": {"input": "$data", "field": "integer_"}},
+                                {"$subtract": [{"$literal": 4}, {"$literal": 1}]},
+                            ]
+                        }
+                    }
+                }
+            ],
+        )
+
+    def test_exact_path(self):
+        with self.assertNumQueries(1) as ctx:
+            self.assertCountEqual(Holder.objects.filter(data__integer=3), [self.objs[3]])
+        query = ctx.captured_queries[0]["sql"]
+        self.assertAggregateQuery(query, "model_fields__holder", [{"$match": {"data.integer_": 3}}])
+
+    def test_lt_expr(self):
+        with self.assertNumQueries(1) as ctx:
+            self.assertCountEqual(
+                Holder.objects.filter(data__integer__lt=Value(4) - 1), self.objs[:3]
+            )
+        query = ctx.captured_queries[0]["sql"]
+        self.assertAggregateQuery(
+            query,
+            "model_fields__holder",
+            [
+                {
+                    "$match": {
+                        "$expr": {
```

Comment:
Curious to the callback chain on this one since the null-check could actually be converted. 

May best be an improvement added later